content/md/articles/cookbooks/date_and_time.md

```diff
@@ -1,13 +1,277 @@
+  [s]
+  (jt/local-date "yyyyddMM" s))
+;; => #'user/ydm
+
+(ydm "20170108")
+;; => #object[java.time.LocalDate 0x4851aa55 "2017-08-01"]
+```
+
+For more formatting patterns check out the
+[DateTimeFormatter](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html)
+class, the table **Predefined Formatters** is where the `:iso-date` came from
+although note it is written as `ISO_DATE` in the **Formatted** column.
+
+More basic recipes can be read in the
+[README](https://dm3.github.io/clojure.java-time/README.html) for
+`clojure.java-time`.
+
+## Date and Time via interop
+
+You can use interop by either using the full package name and the class or by
+importing the class.
+
+Using the package and class name to create an instance of `LocalDate`:
+``` clojure
+(java.time.LocalDate/now) ; => #object[java.time.LocalDate 0x4d98485b "2023-11-01"]
+```
+
+Using the import and the class name to create an instance of `LocalDate`:
+``` clojure
+(import (java.time LocalDate LocalTime LocalDateTime))
+
+;; What's the current day?
+(LocalDate/now)
+;; => #object[java.time.LocalDate 0x28cb30b8 "2023-11-01"]
+;; You may see a different result.
+
+;; What's the current time?
+(LocalTime/now)
+;; => #object[java.time.LocalTime 0x4dc69d04 "12:53:09.791974265"]
+;; You may see a different result.
+
+;; What's the date and time of today?
+(LocalDateTime/now)
+;; => #object[java.time.LocalDateTime 0x6d7119a7 "2023-11-03T12:53:36.765946815"]
+;; You may see a different result.
+
+;; Does date1 come before date2?
+(let [date1 (LocalDate/parse "2023-01-01")
+      date2 (LocalDate/parse "2023-10-01")]
+  (.isBefore date1 date2))
+;; => true
+
+;; Add N days to a date
+(.plus (LocalDate/now) (java.time.Period/ofDays 10))
+;; => #object[java.time.LocalDate 0x507f3b0d "2023-11-13"]
+;; Or
+(.plusDays (LocalDate/now) 10)
+;; => #object[java.time.LocalDate 0x439c6250 "2023-11-13"]
+
+;; What's the date a year before?
+(.minus (LocalDate/parse "2023-11-03") (java.time.Period/ofYears 1))
+;; => #object[java.time.LocalDate 0x37380cf2 "2022-11-03"]
+
+;; Difference in days between a date and a year before
+;; TODO
+;; => 366
+
+;; Return a duration between two dates.
+(java.time.Period/between (LocalDate/parse "2023-11-03")
+                          (.plusYears (LocalDate/parse "2023-11-03")
+                                      1))
+
+;; What day of the week was it?
+(.getDayOfWeek (LocalDate/parse "2023-11-03"))
+;; => #object[java.time.DayOfWeek 0x7823658a "FRIDAY"]
+
+;; How to format the output in a specific way?
+(.format (LocalDate/parse "2023-11-01")
+         (java.time.format.DateTimeFormatter/ofPattern "yyyy/MM/dd"))
+;; => "2023/11/01"
+
+;; To parse the date we just formatted
+(LocalDate/parse "2023/11/01"
+                 (java.time.format.DateTimeFormatter/ofPattern "yyyy/MM/dd"))
+;; => #object[java.time.LocalDate 0xa349bd0 "2023-11-01"]
+
+;; This opens up the door to creating your own parsers.
+;; TODO
+(defn ydm
```

Comment:
This doesn't match the `ydm` shown above which is `yyyyddMM` -- the examples should be consistent.

Reply:
This is a mistake, it should be consistent, I'll correct it.